econml/dml/dml.py

```diff
@@ -1378,181 +1363,25 @@ def ForestDML(model_y, model_t,
-                           cache_values=cache_values,
-                           inference=inference)
-
-    @property
-    def model_final(self):
-        return self._gen_model_final()
-
-    @model_final.setter
-    def model_final(self, model):
-        if model is not None:
-            raise ValueError("Parameter `model_final` cannot be altered for this estimator!")
-
-    @property
-    def n_crossfit_splits(self):
-        return self.n_splits
-
-    @n_crossfit_splits.setter
-    def n_crossfit_splits(self, value):
-        if value != 'raise':
-            warn("Deprecated by parameter `n_splits` and will be removed in next version.")
-        self.n_splits = value
-
-    def shap_values(self, X, *, feature_names=None, treatment_names=None, output_names=None):
-        # SubsampleHonestForest can't be recognized by SHAP, but the tree entries are consistent with a tree in
-        # a RandomForestRegressor, modify the class name in order to be identified as tree models.
-        if self.featurizer_ is not None:
-            F = self.featurizer_.transform(X)
-        else:
-            F = X
-        feature_names = self.cate_feature_names(feature_names)
-        model = copy.deepcopy(self.model_cate)
-        model.__class__ = RandomForestRegressor
-        return _shap_explain_model_cate(self.const_marginal_effect, model, X, self._d_t, self._d_y,
-                                        feature_names=feature_names,
-                                        treatment_names=treatment_names, output_names=output_names)
-    shap_values.__doc__ = LinearCateEstimator.shap_values.__doc__
-
-
-@deprecated("The DMLCateEstimator class has been renamed to DML; "
-            "an upcoming release will remove support for the old name")
-class DMLCateEstimator(DML):
-    pass
-
-
-@deprecated("The LinearDMLCateEstimator class has been renamed to LinearDML; "
-            "an upcoming release will remove support for the old name")
-class LinearDMLCateEstimator(LinearDML):
-    pass
-
-
-@deprecated("The SparseLinearDMLCateEstimator class has been renamed to SparseLinearDML; "
-            "an upcoming release will remove support for the old name")
-class SparseLinearDMLCateEstimator(SparseLinearDML):
-    pass
-
-
-@deprecated("The KernelDMLCateEstimator class has been renamed to KernelDML; "
-            "an upcoming release will remove support for the old name")
-class KernelDMLCateEstimator(KernelDML):
-    pass
-
-
-@deprecated("The NonParamDMLCateEstimator class has been renamed to NonParamDML; "
-            "an upcoming release will remove support for the old name")
-class NonParamDMLCateEstimator(NonParamDML):
-    pass
-
-
-@deprecated("The ForestDMLCateEstimator class has been renamed to ForestDML; "
-            "an upcoming release will remove support for the old name")
-class ForestDMLCateEstimator(ForestDML):
-    pass
+    from . import CausalForestDML
```

Comment:
Is there a reason not to put this import at the top of the file, as usual?

Reply:
(if so, please document why it must be here)

Reply:
It creates a circular import which breaks. BaseDML is used by causalforestdml